src/common.rs

```diff
@@ -16,6 +18,19 @@ pub fn find_seat_price(
     find_seat_price_for_protocol_after_49(stakes, max_number_of_seats, minimum_stake_ratio)
 }
 
+/// With statelessnet feature enabled, we need to take into account `num_chunk_only_producer_seats`.
+pub fn find_max_number_of_seats(protocol_config: &RpcProtocolConfigResponse) -> NumSeats {
+    let seats_before_statelessnet = protocol_config.num_block_producer_seats
+        + protocol_config
+            .avg_hidden_validator_seats_per_shard
+            .iter()
+            .sum::<u64>();
+    std::cmp::max(
+        seats_before_statelessnet,
+        protocol_config.num_chunk_only_producer_seats,
+    )
```

Comment:
I'm not sure if we should sum up these numbers or taking the maximum. Current value is for sure smaller than we need.

```
current  <= maximum <= sum
```
For Statelessnet, the values are 20, 300, 320.

Instead of spending hours on the investigation, I suggest starting using maximum and fix it later again if we see the other issues here.